pkg/util/policy/agent/agent.py

```diff
@@ -78,87 +70,73 @@ def route(self):
+        
+        logging.warning("In do_NP_update : policy_def_valid = %s" % policy_def_valid)
 
-		tenant_vector_chain_exists = TENANT_POLICY_VECTOR_CHAIN in existing_chains
-		logging.info("tenant_vector_chain_exists is %s" % tenant_vector_chain_exists)
+        if policy_def_valid:
+            addr_scheme = policy_def.get("datacenter")
 
-		ALLOW_ANY_VECTOR = "-A %s -j ACCEPT" % TENANT_POLICY_VECTOR_CHAIN
-		logging.info("ALLOW_ANY_VECTOR is %s" % ALLOW_ANY_VECTOR)
+            self.send_response(200)
+            self.wfile.write("Policy definition accepted")
+            logging.warning("Creating policy %s" % policy_def)
 
-		allow_any_vector_exists = ALLOW_ANY_VECTOR in iptables_rules
-		logging.info("allow_any_vector_exists is %s" % allow_any_vector_exists)
+            if self.http_method == "DELETE":
+                logging.warning("Deleting policy %s" % policy_def)
+            else:
+                logging.warning("Creating policy %s" % policy_def)
 
-		filter_idx = iptables_rules.index('*filter')
+            policy_update(addr_scheme, policy_def, delete_policy = ( self.http_method == "DELETE" ))
 
-		if not tenant_vector_chain_exists:
-			logging.info("Tenant policy vector chain does not exist - creating")
-			iptables_rules.insert(filter_idx + 1, ":%s - [0:0]" % TENANT_POLICY_VECTOR_CHAIN)
+            return
+        else:
+            self.send_response(HTTP_Unprocessable_Entity)
+            self.wfile.write(""" Failed to validate policy definition """)
 
-		if allow_any_vector_exists:
-			if isolated:
-				logging.info("Enabling isolation")
-				iptables_rules.remove(ALLOW_ANY_VECTOR)
-		else:
-			if not isolated:
-				logging.info("Disabling isolation")
-				iptables_rules.insert(filter_rules_idx(iptables_rules), ALLOW_ANY_VECTOR)
-		apply_new_ruleset(iptables_rules)
 
-		return
+    def validate_policy(self, policy_def):
+        """
+        Checks that some top level sections are present in the policy.
+        """
+        policy_name = policy_def.get('name')
+        applied_to = policy_def.get('applied_to')
+        rules = policy_def.get('rules')
+        peers = policy_def.get('peers')
+        datacenter = policy_def.get('datacenter')
+        if not policy_name or not applied_to or not rules or not peers or not datacenter:
+            logging.warning("In validate_policy, policy invalid name=%s, applied_to=%s, Rules=%s, datacenter=%s" % (policy_name, applied_to, rules, datacenter))
+            return False
+        return True
 
 
-    def do_POST(self):
+    def decode_request(self):
         """
-        Processes POST requests
-        extracts romana policy definition objects and passes it down for implementation
-    
-        Expected structure: { "method" : "ADDED|DELETED", "policy_definition" : "NP" }
+        Parses json from POST
         """
-
         self.send_header('Content-type','text/html')
         self.end_headers()
         headers = Message(StringIO(self.headers))
```

Comment:
I think line 143 should be moved down one line, so that it is inside of the try-except block. Someone could send a message without the Content-Length header, which would cause the code to crash.

BTW, does the BaseHTTPRequestHandler ensure that the header names are capitalized in a specific way? If someone sends "content-length", will we still get it as "Content-Length"?